client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c

```diff
@@ -471,100 +468,106 @@ kaa_error_t kaa_tcp_channel_set_access_point(void *context, kaa_access_point_t *
 
-    char * connection_data = access_point->connection_data;
+    uint8_t *connection_data = (uint8_t *)access_point->connection_data;
     size_t connection_data_len = access_point->connection_data_len;
-
     size_t position = 0;
-    size_t remaining_to_read = 4;
-    if ((position + remaining_to_read) <= connection_data_len) {
-        channel->access_point.public_key_length = get_uint32_t(connection_data);
-        position += remaining_to_read;
-    } else {
-        KAA_LOG_ERROR(channel->logger, KAA_ERR_INSUFFICIENT_BUFFER, "Kaa TCP channel new access point [0x%08X], "
-                "insufficient connection data length %zu, position %zu",
-                    channel->access_point.id,
-                    connection_data_len,
-                    (position + remaining_to_read));
-        return KAA_ERR_INSUFFICIENT_BUFFER;
+
+    /* Size of remote public key */
+    size_t remaining_to_read = sizeof(uint32_t);
+
+    /* Read remote key length */
+    if ((position + remaining_to_read) > connection_data_len) {
+      goto error_insufficient_buffer;
     }
+    channel->access_point.public_key_length = get_uint32_t(connection_data);
+    position += remaining_to_read;
 
+    /* Read remote public key */
     remaining_to_read = channel->access_point.public_key_length;
-    if ((position + remaining_to_read) <= connection_data_len) {
-        channel->access_point.public_key = (char *) KAA_MALLOC(channel->access_point.public_key_length);
-        KAA_RETURN_IF_NIL(channel->access_point.public_key, KAA_ERR_NOMEM);
-        memcpy(channel->access_point.public_key, connection_data + position, remaining_to_read);
-        position += remaining_to_read;
-    } else {
-        KAA_LOG_ERROR(channel->logger, KAA_ERR_INSUFFICIENT_BUFFER, "Kaa TCP channel new access point [0x%08X], "
-                "insufficient connection data length  %zu, position %zu",
-                    channel->access_point.id,
-                    connection_data_len,
-                    (position + remaining_to_read));
-        return KAA_ERR_INSUFFICIENT_BUFFER;
-    }
-
-    remaining_to_read = 4;
-    if ((position + remaining_to_read) <= connection_data_len) {
-        channel->access_point.hostname_length = get_uint32_t(connection_data + position);
-        position += remaining_to_read;
-    } else {
-        KAA_LOG_ERROR(channel->logger, KAA_ERR_INSUFFICIENT_BUFFER, "Kaa TCP channel new access point [0x%08X], "
-                "insufficient connection data length  %zu, position %zu",
-                    channel->access_point.id,
-                    connection_data_len,
-                    (position + remaining_to_read));
-        return KAA_ERR_INSUFFICIENT_BUFFER;
+    if ((position + remaining_to_read) > connection_data_len) {
+      goto error_insufficient_buffer;
     }
+    channel->access_point.public_key = (uint8_t *) KAA_MALLOC(channel->access_point.public_key_length);
+    if (!channel->access_point.public_key) {
+      return KAA_ERR_NOMEM;
```

Comment:
Error returns mixed with gotos. Please, choose one and consistent way to handle errors